cmd/argo/commands/root.go

```diff
@@ -128,10 +132,33 @@ If your server is behind an ingress with a path (you'll be running "argo server
 		cli.SetLogLevel(logLevel)
 		cmdutil.SetGLogLevel(glogLevel)
 		log.WithField("version", argo.GetVersion()).Debug("CLI version")
+		printVersionMismatchWarning(command)
 	}
 	command.PersistentFlags().StringVar(&logLevel, "loglevel", "info", "Set the logging level. One of: debug|info|warn|error")
 	command.PersistentFlags().IntVar(&glogLevel, "gloglevel", 0, "Set the glog logging level")
 	command.PersistentFlags().BoolVarP(&verbose, "verbose", "v", false, "Enabled verbose logging, i.e. --loglevel debug")
 
 	return command
 }
+
+// printVersionMismatchWarning logs a warning if the CLI version does not match the server version
+func printVersionMismatchWarning(command *cobra.Command) {
+	// if ARGO_SERVER isn't set there's no need to compare server and cli version
+	if _, ok := os.LookupEnv("ARGO_SERVER"); ok {
```

Comment:
Let's invert this conditional so that there is less nesting:
```suggestion
	if _, ok := os.LookupEnv("ARGO_SERVER"); !ok {
	    return
	}
	// .. logic below
```